src/platform/lumin-runtime/elements/builders/image-builder.js

```diff
@@ -130,6 +139,35 @@ export class ImageBuilder extends UiNodeBuilder {
     }
   }
 
+  _createSpinner (prism) {
+    const spinner = this._createNode(ui.UiLoadingSpinner, 'Create', prism, ui.LoadingSpinnerType.k2dSpriteAnimation);
+    const [w, h] = spinner.getSize();
+    const [x, y, z] = spinner.getLocalPosition();
+
+    spinner.setLocalPosition([x - (w / 2), y - (h / 2), z]);
```

Comment:
I think it can be error-prone to change the position of something based on its previous position (unless it's an animation). Do we need [x,y,z] values at all (they should be set to 0 by default)?

```
spinner.setLocalPosition([-0.5 * w, -0.5 * h, 0]);
```

Reply:
I would prefer to adjust the element values instead of hard coded numbers. That way in case the default values from LRT change our code won't need update.